src/aggregator/client/queue.go

```diff
@@ -142,10 +147,27 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 		bufCh:    make(chan protobuf.Buffer, queueSize),
 		doneCh:   make(chan struct{}),
 	}
+
+	var (
+		ticker *time.Ticker
+		tC     <-chan (time.Time)
+	)
+	if batchSize > 0 && minWriteInterval != 0 {
```

Comment:
Perhaps best to always assume > 0 for these consts?  And if they aren't then return an error on construction or something?  Can simplify the code a lot if so.

Reply:
Yeah this is just a PoC without any config plumbing, will fix and adjust (both defaults and init logic).